header-footer-grid/Traits/Core.php

```diff
@@ -90,6 +90,27 @@ public function sanitize_responsive_int_json( $input ) {
 			if ( isset( $filtered[ $key ] ) && is_numeric( $value ) ) {
 				$filtered[ $key ] = (int) $value;
 			}
+
+			// Optional if a suffix is present. We sanitize the array of suffixes below.
+			if ( 'suffix' === $key && is_array( $value ) ) {
+				$filtered['suffix'] = array(
+					'mobile'  => 'px',
+					'tablet'  => 'px',
+					'desktop' => 'px',
+				);
+				foreach ( $value as $device => $suffix ) {
+					if ( ! in_array( $suffix, array( 'px', 'em', 'rem', '%' ), true ) ) {
+						continue;
+					}
+
+					// Additionally; device check was added. (for key check I think we can leverage expected keys from $filtered since we expect 1-1 match for devices between suffix and input devices)
```

Comment:
We can update the comment, I just kept that as inform you about what I want to explain